plugins/woocommerce/client/legacy/js/admin/wc-shipping-zone-methods.js

```diff
@@ -351,6 +351,44 @@
 
 					$( document.body ).trigger( 'init_tooltips' );
 				},
+				// Cost values need to be stripped of thier thousandth separators and made sure
+				// the decimal separator is a ".".
+				unformatShippingMethodNumericValues: function( data ) {
+					if ( ! window.wc.ShippingCurrencyContext ) {
+						return data;
+					}
+
+					const config = window.wc.ShippingCurrencyContext.getCurrencyConfig();
+					const numericValuesFields = [
+						'woocommerce_free_shipping_min_amount',
+						'woocommerce_flat_rate_cost',
+					];
+
+					numericValuesFields.forEach( function( field ) {
+						const formattedValue = data[ field ];
+						
+						if ( ! formattedValue ) {
+							return;
+						}
+
+						if ( 'number' === typeof formattedValue ) {
```

Comment:
I think we should use `Number.isFinite()` to test instead - `NaN` also evalutes to typeof 'number'